ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.java

```diff
@@ -0,0 +1,173 @@
+package com.bugsnag.android.ndk;
+
+import android.os.Build;
+
+import com.bugsnag.android.Configuration;
+import com.bugsnag.android.MetaData;
+import com.bugsnag.android.NativeInterface;
+import com.bugsnag.android.Breadcrumb;
+
+import java.io.File;
+import java.util.List;
+import java.util.Observer;
+import java.util.Observable;
+import java.util.UUID;
+
+/**
+ * Observes changes in the Bugsnag environment, propagating them to the native layer
+ */
+public class NativeBridge implements Observer {
+    static {
+        System.loadLibrary("bugsnag-ndk");
+    }
+
+    public static native void install(String reportingDirectory, boolean autoNotify, int apiLevel);
+    public static native void deliverReportAtPath(String filePath);
+    public static native void addBreadcrumb(String name, String type, String timestamp, Object metadata);
+    public static native void addMetadataString(String tab, String key, String value);
+    public static native void addMetadataDouble(String tab, String key, double value);
+    public static native void addMetadataBoolean(String tab, String key, boolean value);
+    public static native void addHandledEvent();
+    public static native void clearBreadcrumbs();
+    public static native void clearMetadataTab(String tab);
+    public static native void removeMetadata(String tab, String key);
+    public static native void startedSession(String sessionID, String key);
+    public static native void updateAppVersion(String appVersion);
+    public static native void updateBuildUUID(String appVersion);
+    public static native void updateContext(String context);
+    public static native void updateInForeground(boolean inForeground);
+    public static native void updateLowMemory(boolean lowMemory);
+    public static native void updateOrientation(String orientation);
+    public static native void updateMetadata(Object metadata);
+    public static native void updateReleaseStage(String releaseStage);
+    public static native void updateUserId(String newValue);
+    public static native void updateUserEmail(String newValue);
+    public static native void updateUserName(String newValue);
+
+
+    @Override
+    public void update(Observable observable, Object rawMessage) {
+        NativeInterface.Message message;
+        Object arg = null;
```

Comment:
It does, but it will also complain if it isn't initialized as it detects a possible condition where it could be used first (even though it is not the case).

Reply:
I can't see the complaint if that isn't initialised - would you be able to clarify what that is?